src/procedures/utility/build_weighted_matrix.c

```diff
@@ -0,0 +1,454 @@
+				info = GrB_Matrix_apply(M, NULL, NULL, toMatrix, M, NULL);
+			} else {
+				// apply to get the min (max) valued edge from the tensor
+				info = GrB_Matrix_apply_BinaryOp2nd_UDT(M, NULL, NULL,
+						toMatrixMin, M, (void *) (&ctx), NULL);
+				ASSERT(info == GrB_SUCCESS);
+
+			}
+		}
+
+		// add and pick the min (max) valued edge if there is overlap
+		info = GrB_Matrix_eWiseAdd_BinaryOp(_A, NULL, NULL, minID, _A, M, NULL);
+		ASSERT(info == GrB_SUCCESS);
+		GrB_free(&M);
+	}
+
+	//--------------------------------------------------------------------------
+	// compute L
+	//--------------------------------------------------------------------------
+
+	// create vector N denoting all nodes participating in the algorithm
+	if (rows != NULL) {
+		info = GrB_Vector_new(&_N, GrB_BOOL, nrows);
+		ASSERT(info == GrB_SUCCESS);
+	}
+
+	// enforce labels
+	if (n_lbls > 0) {
+		Delta_Matrix DL = Graph_GetLabelMatrix(g, lbls[0]);
+
+		GrB_Matrix L;
+		info = Delta_Matrix_export(&L, DL, GrB_BOOL);
+		ASSERT(info == GrB_SUCCESS);
+
+		// L = L U M
+		for (unsigned short i = 1; i < n_lbls; i++) {
+			DL = Graph_GetLabelMatrix(g, lbls[i]);
+
+			info = Delta_Matrix_export(&M, DL, GrB_BOOL);
+			ASSERT(info == GrB_SUCCESS);
+
+			info = GrB_Matrix_eWiseAdd_Semiring(L, NULL, NULL,
+					GxB_ANY_PAIR_BOOL, L, M, NULL);
+			ASSERT(info == GrB_SUCCESS);
+
+			GrB_Matrix_free(&M);
+		}
+
+		// A = L * A * L
+		info = GrB_mxm(_A, NULL, NULL, GxB_ANY_SECOND_UINT64, L, _A, NULL);
+		ASSERT(info == GrB_SUCCESS);
+
+		info = GrB_mxm(_A, NULL, NULL, GxB_ANY_FIRST_UINT64, _A, L, NULL);
+		ASSERT(info == GrB_SUCCESS);
+
+		// set N to L's main diagonal denoting all participating nodes 
+		if (_N != NULL) {
+			info = GxB_Vector_diag(_N, L, 0, NULL);
+			ASSERT(info == GrB_SUCCESS);
+		}
+
+		// free L matrix
+		info = GrB_Matrix_free(&L);
+		ASSERT(info == GrB_SUCCESS);
+	} else if (rows != NULL) {
+		// no labels, N = [1,....1]
+		info = GrB_Vector_assign_BOOL(
+			_N, NULL, NULL, true, GrB_ALL, nrows, NULL);
+		ASSERT(info == GrB_SUCCESS);
+	}
+
+	if (symmetric) {
+		// make A symmetric A = A + At
+		info = GrB_Matrix_eWiseAdd_BinaryOp(_A, NULL, NULL, minID, _A, _A,
+				GrB_DESC_T1);
+		ASSERT(info == GrB_SUCCESS);
+	}
+
+	n = nrows;
+	if (compact) {
+		// determine the number of nodes in the graph
+		// this includes deleted nodes
+		n = Graph_UncompactedNodeCount(g);
+
+		// get rid of extra unused rows and columns
+		GrB_Info info = GrB_Matrix_resize(_A, n, n);
+		ASSERT(info == GrB_SUCCESS);
+
+		if (rows != NULL) {
+			info = GrB_Vector_resize(_N, n);
+			ASSERT(info == GrB_SUCCESS);
+		}
+	}
+
+	//--------------------------------------------------------------------------
+	// compute weight matrix
+	//--------------------------------------------------------------------------
+
+	if (A_w) {
+		info = GrB_Matrix_new(A_w, GrB_FP64, n, n);
+		ASSERT(info == GrB_SUCCESS);
+
+		if (weight == ATTRIBUTE_ID_NONE) {
+			// if no weight specified, weights are zero
```

Comment:
How come `A_w` is present and `weight` is either wasn't specified or referring to a non existing attribute-id?
I think that in the case the user mentioned a non existing attribute we should error.